src/bin/rav1e.rs

```diff
@@ -7,16 +7,13 @@
 // Media Patent License 1.0 was not distributed with this source code in the
 // PATENTS file, you can obtain it at www.aomedia.org/license/patent.
 
-use y4m;
-
 #[macro_use]
 extern crate scan_fmt;
 
 mod common;
 mod decoder;
-mod muxer;
+mod mux;
```

Comment:
I wouldn't change the name

Reply:
We put the Muxer trait to muxer.rc, clippy will report an error if the directory also named muxer. 
any suggestion for fix this?
thanks

Reply:
The layout I had in mind is:
```
|- ./
   |- decoder/
      |- mod.rs
      |- y4m.rs
   |- muxer/
      |- ivf.rs
      |- mod.rs
      |- mp4.rs
      |- mkv.rs
   |- rav1e.rs
   |- common.rs
```

I just checked and this seems working :)

Reply:
two questions for this. 
1. avformat can support many extensions like .mp4,  .mkv, .webm. Do we really need different files for mp4 and mkv? What is the benefit?
2.  We have two major things in the muxer module. The muxer trait and the create_muxer factory function.  Where we put these two things?
Put them all in mod.rs may introduce cycle dependency. ivf and mp4 depend on mods because of Muxer trait.  mods depend on ivf, mp4 because of create_muxer.

thanks

Reply:
> two questions for this.
> 
>     1. avformat can support many extensions like .mp4,  .mkv, .webm. Do we really need different files for mp4 and mkv? What is the benefit?

We do.

Avformat is not the only provider for muxing, e.g. l-smash is traditionally preferred to mux mp4 when available.

>     2. We have two major things in the muxer module. The muxer trait and the create_muxer factory function.  Where we put these two things?
>        Put them all in mod.rs may introduce cycle dependency. ivf and mp4 depend on mods because of Muxer trait.  mods depend on ivf, mp4 because of create_muxer.

There aren't problems with cycle dependencies in rust.

Reply:
good to know those, I will update according to your comment.
>Avformat is not the only provider for muxing, e.g. l-smash is traditionally preferred to mux mp4 when available.

> There aren't problems with cycle dependencies in rust.

thanks for suggestion

Reply:
Thank you for your work :) I know how frustrating can be refining a patch for something that works well enough for you already.

Reply:
Do not worry. Your request is very reasonable. The way is more clear than my original patch. I am happy to do this.

Reply:
updated, please help review.
thanks